extensions/positron-assistant/src/utils.ts

```diff
@@ -260,22 +298,22 @@ function convertToolResultToAiMessageExperimentalContent(
 					};
 				} else {
 					throw new Error(
-						`Unsupported part type on tool result message: ${(content as any).constructor?.name ?? typeof content}`
+						`Unsupported part type on tool result message: ${typeof content === 'object' && content !== null && 'constructor' in content ? (content.constructor as { name?: string }).name : typeof content}`
 					);
 				}
 			}
 		);
-		toolMessage.content[0].result = toolResultContent;
-		toolMessage.content[0].experimental_content = toolResultContent;
+		toolMessage.content[0].output = { type: 'content' as const, value: toolResultContent };
 	}
 
-	return toolMessage;
+	return toolMessage as ai.ToolModelMessage;
 }
 
 /**
- * Convert a getPlot tool result into a Vercel AI message.
+ * Convert a getPlot tool result into an AI SDK tool message with image content.
+ * AI SDK 5 supports images in tool results via the 'content' output type with 'media' parts.
  */
-function getPlotToolResultToAiMessage(part: vscode.LanguageModelToolResultPart2): ai.CoreUserMessage {
+function getPlotToolResultToAiMessage(part: vscode.LanguageModelToolResultPart2): ai.ToolModelMessage {
```

Comment:
Just wondering: is this function still needed? I think that `convertToolResultToAiMessageExperimentalContent` should be able to handle tool results with images.